test/system/320-system-df.bats

```diff
@@ -2,23 +2,19 @@
 #
 # tests for podman system df
 #
+# DO NOT PARALLELIZE. All of these tests require complete control of images.
+#
 
 load helpers
 
-function setup() {
-    # Depending on which tests have been run prior to getting here, there
-    # may be one or two images loaded. We want only '$IMAGE', not the
-    # systemd one.
-    run_podman rmi -f $SYSTEMD_IMAGE
-
-    basic_setup
-}
-
-function teardown() {
-    basic_teardown
-
-    # In case the active-volumes test failed: clean up stray volumes
+function setup_file() {
+    # Pristine setup: no pods, containers, volumes, images
+    run_podman pod rm -a -f
+    run_podman rm -f -a -t0
     run_podman volume rm -a
+    run_podman image rm -f -a
```

Comment:
I'm sorry, I've tried a few times but still don't understand your concern. You mean the leak check in  `basic_teardown`? I don't see how this could break that?

Reply:
Well basic_teardown doesn't do a leak_check by default since https://github.com/containers/podman/pull/22909.

The only leak check done is in teardown_suite, this means that if a test run before the systemd df tests leaked a container we will not catch it due this change. Granted there are other tests that actual test --all so they behave similar so maybe it is not that big of a deal. And we do use PODMAN_BATS_LEAK_CHECK in nightly runs to catch leaks hopefully anyways

Reply:
Oh I think I see. But: the whole reason for this change is that a pre-320 test _did_ leak a container, and that caused this entire file to fail with lots of noisy and unhelpful errors. I don't remember if the final leak check also caught the leaks or not, because I only focused on the errors in order and probably never made it to the bottom of the log.

One ugly workaround could be to run `leak_check` here, before all the `rm`s. It doesn't appeal to me, in the sense of separation of responsibilities (test code should not have to worry about infrastructure) and because a leak check failure would still be confusing to a reader, but probably less confusing than a bunch of system-df failures. Any ideas on how to do this better?

Reply:
Yeah I am afraid there is no good answer here... I can live with it but if there are leaks the nightly run should catch it already. So if you say you need this because of leaks then this seems to be a concern that we should address and then maybe remove the rm --all here?